lib/tracker/mode_s/decode.go

```diff
@@ -494,9 +494,20 @@ func decodeFlightNumber(b []byte) []byte {
 	callsign[6] = aisCharset[((b[4]&15)<<2)|(b[5]>>6)]
 	callsign[7] = aisCharset[b[5]&63]
 
+	score := 0
+	for i := 0; i < 8; i++ {
+		if (callsign[i] > 'A' && callsign[i] < 'Z') || callsign[i] == '-' || callsign[i] == ' ' || (callsign[i] > '0' && callsign[i] < '9') {
+			score++
+		}
+	}
+	if score < 4 {
+		// do not trust a callsign with fewer than 4 A-Z0-9 chars
+		callsign = nil
+	}
+
 	// because planes have sent us things like A90004A0200000000000007D8DB4
 	// we need
-	if string(callsign) == "@@@@@@@@" {
+	if string(callsign) == "@@@@@@@@" || string(callsign) == "--------" {
```

Comment:
Could we make this "zero allocation" by doing something like this:

At the package level:

```go
var (
    atSigns  = []byte("@@@@@@@@")
    dashes   = []byte("--------")
)
```

...then change the if statement:

```go
if bytes.Equal(callsign, atSigns) || bytes.Equal(callsign, dashes) {
    callsign = nil
}
```

Reply:
We could also add a short-circuit:

```go
if len(callsign) == 8 {
    if callsign[0] == '@' && bytes.Equal(callsign, allAt) ||
       callsign[0] == '-' && bytes.Equal(callsign, allDash) {
        return nil
    }
}
```

Reply:
the []byte <=> string conversion in golang is special cased, there is no alloc for doing this IIRC.

to confirm this I added a benchmark and ran it
```
go test ./lib/tracker/mode_s/ -bench BenchmarkDecodeFlightNumber -benchmem
goos: linux
goarch: amd64
pkg: plane.watch/lib/tracker/mode_s
cpu: AMD Ryzen 9 7900X 12-Core Processor            
BenchmarkDecodeFlightNumber-24          69043420                15.95 ns/op            8 B/op          1 allocs/op
PASS
ok      plane.watch/lib/tracker/mode_s  1.112s
```

that 1/alloc with be for `callsign := make([]byte, 8)`